### PR TITLE
monitoring: enable tracing self instrumentation in APM Server

### PR DESCRIPTION
**Pull request by 1pkg**
## Motivation/summary

This PR addresses the issue https://github.com/elastic/apm-server/issues/14230 and enables self instrumentation in APM Server.

## Checklist

- [x] Update [CHANGELOG.asciidoc](https://github.com/elastic/apm-server/blob/main/CHANGELOG.asciidoc)
- [ ] Documentation has been updated

For functional changes, consider:
- Is it observable through the addition of either **logging** or **metrics**?
- Is its use being published in **telemetry** to enable product improvement?
- Have system tests been added to avoid regression?

## How to test these changes

For the detailed steps to test this changes in self-hosted APM Server refer to https://github.com/elastic/apm-server/pull/13514#issuecomment-2205243038

## Related issues

https://github.com/elastic/apm-server/issues/14230

### CODE REVIEW

**Review by marclop — COMMENTED** [review]
I think this change needs a changelog entry

File: changelogs/head.asciidoc [hunk 1]
```diff
@@ -22,3 +22,4 @@ https://github.com/elastic/apm-server/compare/8.15\...main[View commits]
 ==== Added
 
 - APM Server will no longer retry an HTTP request that returned 502s, 503s, 504s. It will only retry 429s. {pull}13523[13523]
+- Enable self-instrumentation tracing again with transaction sampling support {pull}14231[14231]
```
Comment (marclop) [inline changelogs/head.asciidoc:25]: I think we need to reword the changelog entry to make it actionable, i.e. what does this mean for users? How do they benefit from this change? @axw @simitt how do you envision users will use this and how could we phrase it nicely so it's not cryptic?

Comment (axw) [reply changelogs/head.asciidoc:25]: Hm, well I think ideally the functionality would be documented at https://www.elastic.co/guide/en/fleet/master/elastic-agent-monitoring-configuration.html, and we'd just mention here that we have enabled support for APM instrumentation of APM Server itself when running under Elastic Agent.

In the mean time... I don't know. I think what's written is OK, but yes a little bit cryptic. What about:

> APM Server now supports emitting distributed tracing for its own operation when running under Elastic Agent, and adds support for configuring a sampling rate

**Comment by 1pkg** [comment]
While testing the changes manually, I discovered a bug in EA tracing sampling changes where `sampling_rate` is formatted as a binary (hex) with exponent instead of decimal literal without exponent, this breaks subsequent call in [apm-agent-go](https://github.com/elastic/apm-agent-go/blob/dc6553ec54f0a425c849baf33eb655cafcc9a2e0/config.go#L225) when tracer is initialized with a custom sampling rate.

image

**Comment by 1pkg** [comment]
Raised an issue in the agent repo https://github.com/elastic/elastic-agent/issues/5711

**Review by axw — APPROVED** [review]
Thanks @1pkg!

**Comment by 1pkg** [comment]
Can't be merged yet due to the Cloud configuration missing sensible default for `sampling_rate`.

**Review by simitt — APPROVED** [review]

**Comment by lahsivjar** [comment]
## Testing on ESS

<details>
<summary>Tested successfully with 8.16 BC2, can visualize traces (screenshots attached)</summary>

Screenshot 2024-10-29 at 10 27 37 PM
Screenshot 2024-10-29 at 10 36 18 PM

</details>

## Testing on Standalone

<details>
<summary>Tested successfully with commit SHA ae423f0e4e22643f9c02b489a4fdaeca76bcf26e</summary>

Ran the elastic-agent locally with the following elastic-agent.yml:

```yaml
fleet:
  enabled: true

agent.monitoring:
  # enabled turns on monitoring of running processes
  enabled: true
  traces: true
  apm:
    environment: "816bctesting"
    global_labels:
      foo: localbctesting
    hosts:
      - https://88e3f5<REDACTED>.elastic-cloud.com:443 # this is a personal ESS cluster where I am sending monitoring data
    secret_token: <REDACTED>
    sampling_rate: 1

outputs:
  default:
    type: elasticsearch
    hosts:
      - 'http://elasticsearch:9200'
    username: admin
    password: changeme
```

Observed traces on `88e3f5` (configured monitoring cluster):

Screenshot 2024-10-29 at 11 16 57 PM
Screenshot 2024-10-29 at 11 17 33 PM

</details>

**Pull request merged**